iml-gui/crate/src/page/stratagem/report.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+use crate::{
+    components::{font_awesome, paging, panel, restrict, table},
+    extensions::MergeAttrs as _,
+    generated::css_classes::C,
+    sleep_with_handle, GMsg, RequestExt,
+};
+use futures::channel::oneshot;
+use iml_graphql_queries::{stratagem, Response};
+use iml_wire_types::{GroupType, Session, StratagemReport};
+use number_formatter::format_bytes;
+use seed::{prelude::*, *};
+use std::{cmp::Ordering, time::Duration};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SortField {
+    ModifyTime,
+    Size,
+    Filename,
+}
+
+impl Default for SortField {
+    fn default() -> Self {
+        Self::ModifyTime
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct Model {
+    pager: paging::Model,
+    rows: Vec<StratagemReport>,
+    sort: (SortField, paging::Dir),
+    cancel: Option<oneshot::Sender<()>>,
+}
+
+#[derive(Clone, Debug)]
+pub enum Msg {
+    FetchReports,
+    Reports(fetch::ResponseDataResult<Response<stratagem::list_reports::Resp>>),
+    Page(paging::Msg),
+    Sort,
+    SortBy(table::SortBy<SortField>),
+    DeleteReport(String),
+    ReportDeleted(fetch::ResponseDataResult<Response<stratagem::delete_report::Resp>>),
+    Noop,
+}
+
+pub fn view(model: &Model, session: Option<&Session>) -> Node<Msg> {
+    if model.rows.is_empty() {
+        return empty![];
```

Comment:
Since reports are currently the only thing in here, it would be nice to present a message like no reports found, or something to that effect.